### DISCUSSION TITLE
Cause of warning

### DISCUSSION THREAD

**Discussion by marstonsward** [xgcm/xgcm#585] [Q&A]
When using this example below, I keep getting this warning:

```python
# First create an xgcm grid object
grid = Grid(ds, coords={'Z': {'center':'z'}}, periodic=False)

# define the target values in density, linearly spaced
theta_target = np.linspace(0, 3, 20)

# and transform
phi_transformed = grid.transform(ds.phi, 'Z', theta_target, target_data=ds.theta)
phi_transformed
```
Warning:

```bash
FutureWarning: From version 0.8.0 the Axis computation methods will be removed, in favour of using the Grid computation methods instead. i.e. use `Grid.transform` instead of `Axis.transform`
  warnings.warn(
```
What can I do to fix this?

**Comment by jbusecke** [comment]
You could install the latest master (we have completed the deprecation cycle and the warning is removed, but this is not released yet) or you could temporarily [suppress this warning](https://docs.python.org/3/library/warnings.html).

**Reply by jbusecke** [reply]
To be completely clear. I think this is actually a bug you discovered! Since you used the `grid.transform` method this warning should have never been raised. But I think this is actually fixed with the latest master.

**Reply by marstonsward** [reply]
That is good. I like finding and squashing bugs :-)
I already suppressed the warning. 
I wonder if I could ask another question: 
In the transformation code, if I understand it correctly, the grid is defined with the `Z` as the coordinate to transform and it seems the dataset, ds, must contain both `dataArrays` to be operated on. And the `Z` in the transform code serves the role to tell `xgcm` what axis to operate on, but the new da replaces the `Z` name with the name of the target data. Is there a way to control this? For my input da, `Z name` was `lev' and I would like to keep that name.

**Reply by jbusecke** [reply]
The simplest way to handle this is to rename the dimension afterwards with xarrays `.rename({'oldname', 'newname'})`.
